### ISSUE TITLE
predict and predict_proba give conflicting predictions

### ISSUE THREAD

**Issue by emilioMaddalena** [huggingface/setfit#587]
This is somewhat related to #462.

**The problem:**

If I call `model.predict()`, I get a label. If I call `model.predict_proba()` and try to get the final string via `model.labels` or `model.id2label`, I get a different label.

**Why this happens:**

I think it's because the sklearn model head (where the probabilities are extracted from) might reorganize the labels, leading to an inconsistency between what the model thinks the order is and how the probs are returned.

**Workaround**

Doing `model.model_head.classes_[int(np.argmax(probs))]`. However, I think that's not the intended way since classes_ is a private attribute?

**Comment by emilioMaddalena** [comment]
Some context here. I noticed this tends to happen when you explicitly pass labels to `SetFitModel.from_pretrained()`. If you don't, the model infers them from the data it's given and is free to reorganize them always matching the head, thus the problem is gone. Idk if that's always the case though.

The problem can be reproduced with the code below:

```python
# Training data
sentences = [
    # food
    "I enjoy making homemade pizza on weekends.",
    "Fresh vegetables add color and flavor to any meal.",
    "A freshly baked cake can brighten any celebration.",
    "Many people love discovering new coffee shops.",
    # chemestry
    "Chemistry focuses on the composition, structure, and properties of matter.",
    "Chemical reactions can transform one set of substances into entirely different ones.",
    "Acids and bases react to form salts and water in neutralization processes.",
    "Organic chemistry studies carbon-based compounds essential to living organisms.",
    # sports
    "Regular exercise boosts both health and mood.",
    "Soccer is popular in many countries around the world.",
    "Professional athletes maintain strict training schedules.",
    "A supportive crowd can motivate players to perform better.",
]
labels = [
    "food",
    "food",
    "food",
    "food",
    "chemistry",
    "chemistry",
    "chemistry",
    "chemistry",
    "sports",
    "sports",
    "sports",
    "sports",
]
train_dataset = Dataset.from_dict({"text": sentences, "label": labels})

# Download model and fix classes
model_id = "sentence-transformers/all-MiniLM-L6-v2"
model = SetFitModel.from_pretrained(
    model_id,
    labels=[
        "food",
        "chemistry",
        "sports",
    ],
)

# Fine tune
SEED = 100
random.seed(SEED)
np.random.seed(SEED)
torch.manual_seed(SEED)
args = TrainingArguments(
    batch_size=32,
    num_epochs=3,
)
trainer = Trainer(
    model=model,
    args=args,
    train_dataset=train_dataset,
)
trainer.train()

# 'I enjoy making homemade pizza on weekends.'
sentence = sentences[0]
print(sentence)

# 'food' 
pred = model.predict(sentence)
print(pred)

# 'chemistry' 
probs = model.predict_proba(sentence, as_numpy=True)
pred = model.id2label[int(np.argmax(probs))]
print(pred)

# ['food', 'chemistry', 'sports']
print(model.labels)
# ['chemistry', 'food', 'sports']
print(model.model_head.classes_)
```

**Comment by emilioMaddalena** [comment]
@tomaarsen, if you think it's a good idea, I can work on a fix that would enforce consistency between the two methods.

I'm thinking of either forcing the model head to have labels in exactly the same order as the ones passed by the user, or to modify `predict_proba` to rearrange the probabilities and make them match the `model.labels` order.

What do you think?
